### PR TITLE
Add support for cross tenant and cross subscription restore operations

### PR DESCRIPTION
**Pull request by imadan1996**
## Description
This PR makes the following changes: -

1) Added support for cross restore operations by using optional param **sourceResourceId**.
2) Added support for cross tenant restore operations.
3) Reference the new 2.1.1-preview .NET sdk, as it has the newly added **sourceResourceId** param. [[Microsoft.Azure.Management.Sql_2.1.1-preview](https://www.nuget.org/packages/Microsoft.Azure.Management.Sql/2.1.1-preview)]

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
     - `Get-AzSqlInstanceLink`
     - `Remove-AzSqlInstanceLink`
     - `Set-AzSqlInstanceLink`
+* Added support for cross tenant and cross subscription restore operations for `Restore-AzSqlDatabase`
```
Comment (nadejdapol) [inline src/Sql/Sql/ChangeLog.md:34]: Need to mention that it's only supported for DataWarehouse, say:

Added support for DataWarehouse cross tenant and cross subscription restore operations to `Restore-AzSqlDatabase` cmdlet.

File: src/Sql/Sql/Database Backup/Cmdlet/RestoreAzureRMSqlDatabase.cs [hunk 2]
```diff
@@ -398,7 +399,40 @@ protected override AzureSqlDatabaseModel GetEntity()
...
+            return ModelAdapter.RestoreDatabase(this.ResourceGroupName, restorePointInTime, ResourceId, model, sourceSubscriptionId, auxAuthHeader);
+        }
+
+        /// <summary>
+        /// Parse source subscription id from ResourceId
+        /// </summary>
+        /// <returns>Source Subscription Id</returns>
+        private string ParseSourceSubscriptionIdFromResourceId(string resourceId)
+        {
+            string[] words = resourceId.Split('/');
```
Comment (nadejdapol) [inline src/Sql/Sql/Database Backup/Cmdlet/RestoreAzureRMSqlDatabase.cs:426]: check resourceId for null:
if (string.IsNullOrEmpty(resourceId))
{
    return null;
}

**Review by BethanyZhou — APPROVED** [review]
Looks good to me.

**Pull request merged**